### PR TITLE
feat(web-app): migrate gifplayer from gatsby-theme for mdx usage

### LINKED ISSUE #534: GifPlayer MDX component

### PR DESCRIPTION
**Pull request by francinelucca**
Closes #534 

- [x] Confirm remark-unwrap-images with Alison

#### Testing / reviewing

http://localhost:3000/mdx/gif-player
https://gatsby-theme-carbon.vercel.app/components/GifPlayer/

### CODE REVIEW

File: services/web-app/components/gif-player/gif-player.module.scss [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  width: 1.5rem;
+  height: 1.5rem;
+  padding: 0;
+  border: none;
+  background: transparent;
+  outline: none;
+}
+
+.controls svg {
+  fill: #ffffff;
```
Comment (alisonjoseph) [inline services/web-app/components/gif-player/gif-player.module.scss:20]: Lets replace this with a token, I believe `$icon-inverse` would be appropriate here.

File: services/web-app/components/gif-player/gif-player.module.scss [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+.dark svg {
+  fill: #000000;
```
Comment (alisonjoseph) [inline services/web-app/components/gif-player/gif-player.module.scss:24]: `$icon-primary` I think makes sense here

File: services/web-app/components/gif-player/gif-player.module.scss [hunk 3]
```diff
@@ -0,0 +1,54 @@
+@use '@carbon/react/scss/theme' as theme;
+
+.container {
+  position: relative;
+}
+
+.controls {
+  position: absolute;
+  bottom: 1rem;
+  left: 1rem;
```
Comment (alisonjoseph) [inline services/web-app/components/gif-player/gif-player.module.scss:10]: Do we want spacing tokens for bottom/left?

File: services/web-app/components/gif-player/gif-player.module.scss (hunk 1 above)
Comment (francinelucca) [reply services/web-app/components/gif-player/gif-player.module.scss]: Done!

File: services/web-app/components/gif-player/gif-player.module.scss (hunk 2 above)
Comment (francinelucca) [reply services/web-app/components/gif-player/gif-player.module.scss]: Done!

File: services/web-app/components/gif-player/gif-player.module.scss (hunk 3 above)
Comment (francinelucca) [reply services/web-app/components/gif-player/gif-player.module.scss]: Done! changed for corresponding spacing tokens

**Review by alisonjoseph — APPROVED** [review]
LGTM! I tested the dark prop locally and confirmed it works as expected.

**Review by alisonjoseph — COMMENTED** [review]
Suggestion, should we use the theme component instead of the "dark" class, I did this for the article card and do-dont components. Theme wasn't a thing yet when we built the Gatsby components.

`<Theme theme={color === 'dark' && 'g100'}>`

File: services/web-app/components/gif-player/gif-player.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import {
+  PauseOutline24,
+  PauseOutlineFilled24,
+  PlayOutline24,
+  PlayOutlineFilled24
+} from '@carbon/icons-react'
+import cx from 'classnames'
```
Comment (alisonjoseph) [inline services/web-app/components/gif-player/gif-player.js:13]:
```suggestion
import clsx from 'clsx'
```

File: services/web-app/components/gif-player/gif-player.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+const GifPlayer = ({ children, color, className, isInDialog }) => {
+  const [paused, setPaused] = useState(false)
+
+  const [hovering, setHovering] = useState(false)
+  const onClick = (e) => {
+    e.stopPropagation()
+    setPaused(!paused)
+  }
+
+  const controlsClassNames = cx(styles.controls, {
```
Comment (alisonjoseph) [inline services/web-app/components/gif-player/gif-player.js:35]:
```suggestion
  const controlsClassNames = clsx(styles.controls, {
```

File: services/web-app/components/gif-player/gif-player.js [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    [styles.dark]: color === 'dark'
+  })
+
+  const containerClassNames = cx(className, styles.container, {
```
Comment (alisonjoseph) [inline services/web-app/components/gif-player/gif-player.js:39]:
```suggestion
  const containerClassNames = clsx(className, styles.container, {
```

File: services/web-app/components/gif-player/gif-player.js [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+    [styles.gifInDialog]: isInDialog
+  })
+
+  const staticImageClassNames = cx(styles.imgHidden, {
```
Comment (alisonjoseph) [inline services/web-app/components/gif-player/gif-player.js:43]:
```suggestion
  const staticImageClassNames = clsx(styles.imgHidden, {
```

File: services/web-app/components/gif-player/gif-player.js [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+    [styles.imgDisplayed]: paused
+  })
+
+  const gifClassNames = cx(styles.gifDisplayed, {
```
Comment (alisonjoseph) [inline services/web-app/components/gif-player/gif-player.js:47]:
```suggestion
  const gifClassNames = clsx(styles.gifDisplayed, {
```

**Review by alisonjoseph — CHANGES_REQUESTED** [review]
use clsx instead of cx

**Review by alisonjoseph — APPROVED** [review]

**Review by andreancardona — APPROVED** [review]
Clarified on first render _flashing_ with. @francinelucca all looks good!

**Pull request merged**